src/utilities/__tests__/extendSchema-test.js

```diff
@@ -801,6 +801,107 @@ describe('extendSchema', () => {
     `);
   });
 
+  it('extends interfaces by adding new fields', () => {
+    const ast = parse(`
+      extend interface SomeInterface {
+        newObject: NewObject
+        newInterface: NewInterface
+        newUnion: NewUnion
+        newScalar: NewScalar
+        newTree: [Foo]!
+        newField(arg1: String, arg2: NewEnum!): String
+      }
+
+      type NewObject implements NewInterface {
+        baz: String
+      }
+
+      type NewOtherObject {
+        fizz: Int
+      }
+
+      interface NewInterface {
+        baz: String
+      }
+
+      union NewUnion = NewObject | NewOtherObject
+
+      scalar NewScalar
+
+      enum NewEnum {
+        OPTION_A
+        OPTION_B
+      }
+    `);
+    const originalPrint = printSchema(testSchema);
+    const extendedSchema = extendSchema(testSchema, ast);
+    expect(extendedSchema).to.not.equal(testSchema);
+    expect(printSchema(testSchema)).to.equal(originalPrint);
+    expect(printSchema(extendedSchema)).to.equal(dedent`
+      type Bar implements SomeInterface {
+        name: String
+        some: SomeInterface
+        foo: Foo
```

Comment:
Probably bug here, should fail because both `Foo` and `Bar` are missing additional fields:
```
 newObject: NewObject
 newInterface: NewInterface
 newUnion: NewUnion
 newScalar: NewScalar
 newTree: [Foo]!
 newField(arg1: String, arg2: NewEnum!): String
```

Reply:
Well `printSchema` should not fail here: we should be able to parse, print and extend invalid schemas. Only `validateSchema(extendedSchema)` should fail.

But if that's what you're testing, you should be explicit "can parse and print schema extension with missing Object fields", then prove that it's invalid by testing validateSchema fails (don't test the actual error message though, as we don't wan to force people to update this test when they update that error message).

Reply:
@mjmahone You are right, I completely forget about that.
But I still think this test should contain valid SDL examples to not confuse readers.

> But if that's what you're testing, you should be explicit "can parse and print schema extension with missing Object fields", then prove that it's invalid by testing validateSchema fails (don't test the actual error message though, as we don't wan to force people to update this test when they update that error message).

Totally make sense to add one such test for `buildShema` and `extendSchema` functions 👍